addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -447,16 +438,14 @@ export function createRelatedModels(modelDefs, modelClasses = {}, opts = {}) {
     }
 
     const disabler = new TrapDisabler();
-    function withoutProxyTrap(fn) {
-        return (...args) => disabler.call(fn, ...args);
-    }
-
     const setTrapsCache = {};
-    function instantiateModel(model, { models, records }) {
-        const fields = getFields(model);
-        const Model = modelClasses[model] || Base;
-        if (!(model in setTrapsCache)) {
-            setTrapsCache[model] = function setTrap(target, prop, value, receiver) {
+
+    function instantiate(model) {
```

Comment:
Maybe call it `newRecord` or `createNewRecord`something like that ? Its just an advice, you can keep it like that if you want :-)